src/components/Widgets/Widgets.tsx

```diff
@@ -1,15 +1,22 @@
+import { ChainId } from '@lifi/types';
 import type { WidgetSubvariant } from '@lifi/widget';
 import { Grid, useTheme } from '@mui/material';
 import { useCallback, useLayoutEffect, useMemo, useState } from 'react';
-import { OnRamper, TestnetAlert, Widget } from 'src/components';
+import { OnRamper, SolanaAlert, TestnetAlert, Widget } from 'src/components';
 import { LinkMap, TabsMap } from 'src/const';
-import { useActiveTabStore, useSettingsStore } from 'src/stores';
+import {
+  useActiveTabStore,
+  useChainTokenSelectionStore,
+  useSettingsStore,
+} from 'src/stores';
 import type { StarterVariantType } from 'src/types';
 import { WidgetEvents } from './WidgetEvents';
 import { WidgetContainer } from './Widgets.style';
 
 export function Widgets() {
   const { activeTab, setActiveTab } = useActiveTabStore();
+  const { sourceChainToken, destinationChainToken } =
+    useChainTokenSelectionStore();
```

Comment:
Does this part re-render the widget in response to the event? Please see here https://github.com/lifinance/jumper.exchange/pull/791 why it is bad and restructure the logic if that's needed 🙂

Reply:
THANK YOU! Good point: Check latest commit [refactor: prevent widget re-render](https://github.com/lifinance/jumper.exchange/pull/807/commits/e5d26e9755f697da2ecd890876e44acbc9dee0e8)